### ISSUE TITLE
[BUG] Worker-scope fixture setup skewing test duration

### ISSUE THREAD

**Issue by raulparada** [microsoft/playwright#15338]
Hi! 👋 

**Context:**
- Playwright Version: 1.22.2
- Operating System: macOS 12.1 Monterey
- Node.js version: v18.4.0
- Browser: Chrome
- Extra: I doubt the environment plays a role

**Describe the bug**
The duration of any worker-scoped fixture setup is attributed to the first test that the worker runs afterwards.
i.e. if there's a worker-scoped fixture that takes 20s to setup, and the worker runs 3 tests that each take 5 seconds, the report will say that:
1. test 1 (25s) -> so: counting fixture setup.
2. test 2 (5s)
3. test 3 (5s)

This:
a) greatly skews the metrics for tests duration
b) is non-deterministic, as the fixture setup is not done consistently previous to the same test across runs (if e.g. there's a test failure and the worker restarts)

IMO, duration of worker-scoped fixtures setup should be reported (still, but) completely outside of individual test duration, that'd be my user expectation.

**Comment by AndrewFeeney** [comment]
Yes, I've been trying to understand this myself. This doesn't seem to be in line with the documented behaviour here: https://playwright.dev/docs/test-timeouts#fixture-timeout but I've been trying to figure out if I'm just configuring this wrong.

**Comment by AndrewFeeney** [comment]
It looks to me as though https://github.com/microsoft/playwright/pull/15117 which has been merged, but is awaiting a tagged release, might fix this issue. I haven't been able to test to confirm though.

**Comment by dgozman** [comment]
This is explained in [test timeout](https://playwright.dev/docs/test-timeouts#test-timeout) section - all fixtures are included in the test time. However, you can opt-out of this behavior for any fixture by setting a separate `timeout` for it, as explained in [this section](https://playwright.dev/docs/test-timeouts#fixture-timeout). Let me know whether that works for you.

**Comment by raulparada** [comment]
Hi and thanks @dgozman :)
But issue is not about test timeouts, it's about reported test durations; i.e. my tests are not hitting timeouts, it's _just_ that the runner is attributing the duration of the fixture set up to the first test handled by runner afterwards. Code for my minimal example on description here:

```ts
import { test as baseTest } from "@playwright/test";

const wait = (m: number) => new Promise((r) => setTimeout(r, m));

const test = baseTest.extend<
  {},
  {
    workerFixture: any;
  }
>({
  workerFixture: [
    async ({}, use) => {
      console.log("wait for 20 seconds");
      await wait(20000);
      await use({});
    },
    { scope: "worker" },
  ],
});

test("wait for five seconds, 1", async ({ workerFixture }) => {
  console.log("wait for 5 seconds");
  await wait(5000);
});
test("wait for five seconds, 2", async ({ workerFixture }) => {
  console.log("wait for 5 seconds");
  await wait(5000);
});
test("wait for five seconds, 3", async ({ workerFixture }) => {
  console.log("wait for 5 seconds");
  await wait(5000);
});
```
Output:
```
Running 3 tests using 1 worker

  ✓  tests/test.issue.spec.ts:21:1 › wait for five seconds, 1 (25s)
wait for 20 seconds
wait for 5 seconds
  ✓  tests/test.issue.spec.ts:25:1 › wait for five seconds, 2 (5s)
wait for 5 seconds
  ✓  tests/test.issue.spec.ts:29:1 › wait for five seconds, 3 (5s)
wait for 5 seconds
```

Again, not a nag as this skews the stats as described on summary. I could have a short test once in a while reporting as taking way longer, thus I lose visibility on which tests I need to optimise. Thanks again and have a nice day, cheers.

**Comment by raulparada** [comment]
...and tested with your comment to see if I misunderstood, so changed to:
```ts
...
    { scope: "worker", timeout: 21000 },
...
```
...to see if that would 'isolate' the _duration_ as well as the timeout, but the output was the exact same, so my issue is, I believe, valid

**Comment by AndrewFeeney** [comment]
I've also experienced what @raulparada has experienced. All of my fixtures are worker scoped and have separate timeouts, but it makes no difference to the reported duration of the first tests run by the worker, which still includes the fixture time.

As I already mentioned I'm hoping that the changes introduced in https://github.com/microsoft/playwright/pull/15117 address this. They have been merged into main, but not yet tagged in a release. I haven't had success installing the latest commit locally to test. Evidently my package.json skills are lacking in this department!
